File: frontend/src/lib/conversation-contracts.ts
import { z } from "zod";

export const SkippedPurgeSchema = z.object({
  id: z.string(),
  reason: z.string(),
});
export type SkippedPurge = z.infer<typeof SkippedPurgeSchema>;

export const PurgeAllResponseSchema = z.object({
  purged: z.array(z.string()),
  skipped: z.array(SkippedPurgeSchema),
});
export type PurgeAllResponse = z.infer<typeof PurgeAllResponseSchema>;

File: partyline/conversation_contracts.py
"""Named contracts for conversation operations."""

from pydantic import BaseModel, Field


class SkippedPurge(BaseModel):
    id: str
    reason: str


class PurgeAllResponse(BaseModel):
    purged: list[str] = Field(default_factory=list)
    skipped: list[SkippedPurge] = Field(default_factory=list)


class RetirementBlocker(BaseModel):
    code: str
    message: str


class BlockedArchiveResponse(BaseModel):
    """A refused retirement: one summary line plus every blocker behind it."""

    detail: str
    blockers: list[RetirementBlocker] = Field(default_factory=list)
